requirements-dev.txt

```diff
@@ -1,30 +1,4 @@
--r requirements.txt
-
-# Test & lint packages
-coverage==5.0.3
-mock==3.0.5
-placebo==0.9.0
-flake8==3.7.9
-pytest
-pytest-xdist
-pytest-cov==2.8.1
-twine
-tox==3.14.3
-six >= 1.12.0
-
-# Optional feature testing in c7n
-psutil==5.6.7
-aws_xray_sdk==2.4.3
-importlib_metadata==1.4.0
-
--r tools/c7n_mailer/requirements.txt
--r tools/c7n_azure/requirements.txt
--r tools/c7n_gcp/requirements.txt
--r tools/c7n_kube/requirements.txt
-
-# Specific to testing mailer
-fakeredis==1.1.0
-
+-e .
```

Comment:
Devs will need to be using `poetry` to install right?  So is `requirements-dev.txt` still valuable?  

Should we just do `requirements-ci.txt` instead, and remove the `-e` to fix caching issues with the egg folders?

Reply:
devs only need to use poetry if they are adjusting the dependency graph. existing user integrations with custodian trunk will continue to work out of the box the same way.

we want -e for ci, else we're testing old versions/revisions of local packages from the cache key .. however the caching with egg folders / -e is an independent issue then this pr, and per my last comment on the azure functions pr its also something that can be handled differently by the azure mailer code, ie we have one test out 2.4k that needs it, and i don't think it really does need it.

Reply:
Makes sense.